report/generator/file-namer.js

```diff
@@ -35,11 +35,17 @@ function getFilenamePrefix(name, fetchTime) {
 
 /**
  * Generate a filenamePrefix of hostname_YYYY-MM-DD_HH-MM-SS.
- * @param {{finalUrl: string, fetchTime: string}} lhr
+ * @param {{finalDisplayedUrl: string, finalUrl?: string, fetchTime: string}} lhr
  * @return {string}
  */
 function getLhrFilenamePrefix(lhr) {
-  const hostname = new URL(lhr.finalUrl).hostname;
+  // If LHR is older than 10.0 it will not have the `finalDisplayedUrl` property.
+  // Old LHRs should have the `finalUrl` property which will work fine for the report.
+  // TODO: Move this logic to a shared dependency between file namer and report utils.
```

Comment:
FWIW this file is the way it is because it's a shared dependency between cli and report utils. Zeno's shared deps? :)

It only needs `{{finalDisplayedUrl: string, fetchTime: string}}` and in the report/viewer is only called in `report-ui-features` and `github-api`. What about those callsites call `Util.getFinalDisplayedUrl()` and keep it out of `file-namer`?

Reply:
Went with this suggestion